src/peerstate.rs

```diff
@@ -189,33 +184,18 @@ impl<'a> Peerstate<'a> {
                 res.prefer_encrypt = EncryptPreference::from_i32(row.get(3)?).unwrap_or_default();
                 res.gossip_timestamp = row.get(5)?;
 
-                res.public_key_fingerprint = row.get(7)?;
-                if res
-                    .public_key_fingerprint
-                    .as_ref()
-                    .map(|s| s.is_empty())
-                    .unwrap_or_default()
-                {
-                    res.public_key_fingerprint = None;
-                }
-                res.gossip_key_fingerprint = row.get(8)?;
-                if res
-                    .gossip_key_fingerprint
-                    .as_ref()
-                    .map(|s| s.is_empty())
-                    .unwrap_or_default()
-                {
-                    res.gossip_key_fingerprint = None;
-                }
-                res.verified_key_fingerprint = row.get(10)?;
-                if res
-                    .verified_key_fingerprint
-                    .as_ref()
-                    .map(|s| s.is_empty())
-                    .unwrap_or_default()
-                {
-                    res.verified_key_fingerprint = None;
-                }
+                res.public_key_fingerprint = match row.get::<_, Option<String>>(7)? {
+                    None => None,
+                    Some(s) => Some(s.parse::<Fingerprint>()?),
+                };
```

Comment:
It's tempting to want to write this using and `.map()` but because the `.parse()` can fail we can not fail out of the closure of the map.  Thus we end up with this match with a `None => None` branch, which is not ideal.

But at least it looks better than what it replaces ;)

Reply:
You can use [`transpose`](https://doc.rust-lang.org/std/option/enum.Option.html#method.transpose) to write this nicer.

```rust
let x = row.get::<_, Option<String>>(7)?.map(|s| s.parse::<Fingerprint>()).transpose()?;
```

should do the trick.